rust/datafusion/src/physical_plan/functions.rs

```diff
@@ -523,6 +544,7 @@ fn signature(fun: &BuiltinScalarFunction) -> Signature {
         BuiltinScalarFunction::NullIf => {
             Signature::Uniform(2, SUPPORTED_NULLIF_TYPES.to_vec())
         }
+        BuiltinScalarFunction::RegexpExtract => Signature::Any(3),
```

Comment:
I am not sure this is correct; isn't the signature `Variant([[Utf8, Utf8, UInt64], [LargeUtf8, Utf8, UInt64]])` or something like that?

Note that these signatures are very important because they are used for type validation during logical planning, as well as type coercion at physical planning. Whenever we write `Any`, the logical planning will accept any type. Worse, the type coercer will not perform any coercion.

In this case, because we downcast arg[2] to `Int64Array`, if the user passes a `Int32Array`, the execution panics.

Reply:
Done - I had to use `Int64` instead of `UInt64` because I got errors from `tests/sql.rs`. I have to check how to define a literal as unsigned from within the query string.